packages/plugin/webpack/src/WebpackConfig.ts

```diff
@@ -163,114 +180,122 @@ export default class WebpackConfigGenerator {
-          filename: 'preload.js',
-        },
-        node: {
-          __dirname: false,
-          __filename: false,
-        },
+  buildRendererBaseConfig(target: RendererTarget): webpack.Configuration {
+    return {
+      target: rendererTargetToWebpackTarget(target),
+      devtool: this.rendererSourceMapOption,
+      mode: this.mode,
+      output: {
+        path: path.resolve(this.webpackDir, 'renderer'),
+        filename: '[name]/index.js',
+        globalObject: 'self',
+        ...(this.isProd ? {} : { publicPath: '/' }),
       },
-      rendererConfig || {},
-      { target: 'electron-preload' }
-    );
+      node: {
+        __dirname: false,
+        __filename: false,
+      },
+      plugins: [new AssetRelocatorPatch(this.isProd, !!this.pluginConfig.renderer.nodeIntegration)],
+    };
   }
 
-  async getRendererConfig(entryPoints: WebpackPluginEntryPoint[]): Promise<Configuration[]> {
+  async buildRendererConfig(entryPoints: WebpackPluginEntryPoint[], target: RendererTarget): Promise<Configuration | null> {
+    const entry: webpack.Entry = {};
+
+    if (entryPoints.length === 0) {
+      return null;
+    }
+    const baseConfig: webpack.Configuration = this.buildRendererBaseConfig(target);
     const rendererConfig = await this.resolveConfig(this.pluginConfig.renderer.config);
 
-    return entryPoints.map((entryPoint) => {
-      const baseConfig: webpack.Configuration = {
-        target: this.rendererTarget(entryPoint),
-        devtool: this.rendererSourceMapOption,
-        mode: this.mode,
+    if (target === RendererTarget.Web || target === RendererTarget.ElectronRenderer) {
+      for (const entryPoint of entryPoints) {
+        entry[entryPoint.name] = (entryPoint.prefixedEntries || []).concat([isPreloadOnly(entryPoint) ? entryPoint.preload.js : entryPoint.js]);
+      }
+      const output = {
+        path: path.resolve(this.webpackDir, 'renderer'),
+        filename: '[name]/index.js',
+        globalObject: 'self',
+        ...(this.isProd ? {} : { publicPath: '/' }),
+      };
+      const plugins: webpack.WebpackPluginInstance[] = [];
+      for (const entryPoint of entryPoints) {
+        if (isLocalWindow(entryPoint)) {
+          plugins.push(
+            new HtmlWebpackPlugin({
+              title: entryPoint.name,
+              template: entryPoint.html,
+              filename: `${entryPoint.name}/index.html`,
+              chunks: [entryPoint.name].concat(entryPoint.additionalChunks || []),
+            }) as WebpackPluginInstance
+          );
+        }
+      }
+      return webpackMerge(baseConfig, { entry, output, plugins }, rendererConfig || {});
+    } else if (target === RendererTarget.ElectronPreload || target === RendererTarget.SandboxedPreload) {
+      for (const entryPoint of entryPoints) {
+        entry[entryPoint.name] = (entryPoint.prefixedEntries || []).concat([isPreloadOnly(entryPoint) ? entryPoint.preload.js : entryPoint.js]);
```

Comment:
Why is `.js` here, neither `ElectronPreload` nor `SandboxedPreload` should be referencing anything **other** than the preload JS file